Modules/socketmodule.c

```diff
@@ -546,23 +547,34 @@ typedef struct _socket_state {
        by this module (but not argument type or memory errors, etc.). */
     PyObject *socket_herror;
     PyObject *socket_gaierror;
+} socket_state;
+
+/* Default timeout for new sockets */
+static PyTime_t defaulttimeout = _PYTIME_FROMSECONDS(-1);
```

Comment:
This should be part of the module state, no?

Reply:
It was previously moved to module state because of isolation of subinterpreters and that time we didn't have atomic support in cpython so moving to the state was the best thing. Now we have atomics support so using it as global using atomics solves it for both so I moved it as it was before.

Reply:
I changed it back to module state, as removed the macros as you suggested to always use atomics.